scanpy/tests/test_pca.py

```diff
@@ -153,3 +157,42 @@ def test_pca_n_pcs(pbmc3k_normalized):
     assert np.allclose(
         original.obsp["distances"].toarray(), renamed.obsp["distances"].toarray()
     )
+
+
+def test_pca_mask(array_type, float_dtype):
+
+    pbmc = sc.datasets.pbmc68k_reduced()
+
+    # Test highly_variable ValueError
+    adata = array_type(A_list).astype('float32')
+    with pytest.raises(ValueError):
+        sc.pp.pca(adata, use_highly_variable=True)
+
+    # check warning on mask length
+    mask = np.random.choice([True, False], pbmc.shape[1] + 1)
+    with pytest.raises(ValueError):
+        sc.pp.pca(pbmc, mask=mask, copy=True, dtype=float_dtype)
+
+    # Test if pca result is equal when given mask vs. given use_highly_variable=True
+    mask = pbmc.var['highly_variable']
+    adata_mask = sc.pp.pca(pbmc, mask=mask, copy=True, dtype=float_dtype)
```

Comment:
It would be good to have tests for masks under different keys, since this could work if the array wasn't actually passed. Instead, maybe something like:

```python
pbmc = datasets.pbmc3k_processed.raw.to_adata()
mask = np.random.choice([True, False], pbmc.shape[1] + 1)
pbmc_w_mask = pbmc.copy()
pbmc_w_mask.var["mask"] = mask

sc.pp.pca(pbmc, mask=mask)
sc.pp.pca(pbmc_w_mask, mask="mask")
```

Then compare the values of `pbmc` and `pbmc_w_mask`.